app/lib/utils.dart

```diff
@@ -51,3 +51,56 @@ String canonicalizeVersion(String version) {
   }
   return canonicalVersion.toString();
 }
+
+RegExp _identifierExpr = new RegExp(r'^[a-zA-Z0-9_]*$');
+RegExp _startsWithLetterOrUnderscore = new RegExp(r'^[a-zA-Z_]');
```

Comment:
Consider always using `final` if you don't modify variables (also local variables)